assets/lambda/ImageOptimization.ts

```diff
@@ -0,0 +1,160 @@
+// Taken from: https://github.com/sladg/nextjs-lambda/blob/master/lib/standalone/image-handler.ts
+// There are other open source MIT libraries we can pick, but this seems the most straightforward
+
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3'
+import type { APIGatewayProxyEventV2, APIGatewayProxyStructuredResultV2 } from 'aws-lambda'
+import { IncomingMessage, ServerResponse } from 'http'
+import { defaultConfig, NextConfigComplete } from 'next/dist/server/config-shared'
+import { imageOptimizer as nextImageOptimizer, ImageOptimizerCache } from 'next/dist/server/image-optimizer'
+import { NextUrlWithParsedQuery } from 'next/dist/server/request-meta'
+import { ImageConfigComplete, ImageConfig } from 'next/dist/shared/lib/image-config'
+import { Readable } from 'stream'
+import https from 'https'
+
+const sourceBucket = process.env.S3_SOURCE_BUCKET ?? undefined
+
+/**
+	* The imageConfig object is stringified and passthrough the lambda as an environment variable
+	*/
+const _loadImageConfig = (): ImageConfig => {
+	let imageConfig: ImageConfig = {}
+	const imageConfigString = process.env.NEXT_IMAGE_CONFIG
+	try {
+		imageConfig = JSON.parse(imageConfigString || '{}') as ImageConfig
+	} catch (err) {
+		console.error('Error parsing image config: ', { err, imageConfigString })
+		throw err
+	}
+
+	return imageConfig
+}
+
+const _getRequest = (url: NextUrlWithParsedQuery): Promise<Buffer> => {
+	return new Promise((resolve, reject) => {
+		const req = https.get(url, res => {
+			const chunks: Buffer[] = []
+			res.on('data', chunk => chunks.push(chunk))
```

Comment:
maybe we shouldn't buffer the response here

Reply:
I'll refactor this to stream.